corehq/apps/api/odata/tests/utils.py

```diff
@@ -0,0 +1,72 @@
+from __future__ import absolute_import
+from __future__ import unicode_literals
+
+import base64
+
+from django.test.client import Client
+from django.urls import reverse
+
+from corehq.apps.domain.models import Domain
+from corehq.apps.users.models import WebUser
+
+
+class OdataTestMixin(object):
+
+    view_urlname = None
+
+    @classmethod
+    def setUpClass(cls):
+        super(OdataTestMixin, cls).setUpClass()
```

Comment:
What does this do? This would call `TestCase`/`SimpleTestCase` if it was inheriting from that but this class is inheriting from `object` which does not have that. It also makes it hard to reason about what `super().setUpClass` would do in a class htat inherits from both this and `TestCase`

Reply:
```super().setUpClass``` would look for a ```setUpClass``` method in ```object```, find none, and then look for a ```setUpClass``` method in ```TestCase```, which would be found and executed.

Reply:
Do you have an alternative in mind?  Like, creating a library with functions that perform this common setup steps?

Reply:
Offlined with Nick, but suggested something like:

```
class TestMixin(object):
   def test_a(self):
     pass

class OdataTests(TestCase):
   def setupClass():
        pass

class RealTest(OdataTests, TestMixin):
   def specific_tests():
        pass
```

This ensures that `test_a` does not get run on its own (only when included in a class via the mixin) and that there isn't a diamond inheritance pattern with `setupClass` etc which can be hard to debug: https://github.com/dimagi/commcare-hq/pull/7146#issuecomment-112973367

> super().setUpClass would look for a setUpClass method in object, find none, and then look for a setUpClass method in TestCase, which would be found and executed.

@nickpell I think this means that it would not work if you accidentally swapped the mixin order which can be hard to debug/easy to mess up